pdf_build_src/process_markdowns.py

```diff
@@ -141,6 +141,164 @@ def modify_changelog():
         file.writelines(data)
 
 
+"""Number of chars maximal in one line approximated from a line of the PDF"""
+NB_CHARS_LINE_PDF = 100
+
+def correct_table(table):
+    """Compute the max number of characters and dashes for each column and create the corrected table
+
+    Parameters
+    ----------
+    table : List of List of str
+        Table content extracted from the markdown file.
+
+    Returns
+    -------
+    new_table : List of List of str
+        List of corrected lines of the input table with corrected number of dashes and aligned fences.
+        To be later join with |'s
+    """
+    import numpy as np
+    
+    nb_of_rows = len(table)
+    nb_of_cols = len(table[0])
+
+    nb_of_chars = []
+    for i, row in enumerate(table):
+         # Ignore number of dashes in the count of characters
+        if i != 1:
+            nb_of_chars.append([len(elem) for elem in row])
+
+    # Convert the list to a numpy array and computes the maximum number of chars for each column
+    nb_of_chars_arr = np.array(nb_of_chars)
+    max_chars_in_cols = nb_of_chars_arr.max(axis=0)
+
+    # Computes number of dashes based on the maximal number of characters in each column
+    nb_of_dashes = max_chars_in_cols
+    prop_of_dashes = nb_of_dashes / nb_of_dashes.sum()
+    nb_of_chars_in_pdf = prop_of_dashes * int(NB_CHARS_LINE_PDF)
+
+    # print('    - Number of chars in table cells: {}'.format(max_chars_in_cols))
+    # print('    - Number of dashes (per column): {}'.format(nb_of_dashes))
+    # print('    - Proportion of dashes (per column): {}'.format(prop_of_dashes))
+    # print('    - Number of chars max in column (PDF): {}'.format(nb_of_chars_in_pdf))
+
+    # Offset that can be used to ajust the correction of number of dashes in the first and 
+    # second columns by the number specified
+    offset = [15, 5]
```

Comment:
where do these numbers come from?

Reply:
This is indeed really not trivial to readjust the tables and these offsets are very useful in adjusting columns where a few or a lot of characters are present.